core/src/main/java/org/jruby/ir/IRBuilder.java

```diff
@@ -336,12 +337,15 @@ public void addArgumentDescription(ArgumentType type, RubySymbol name) {
     public void addInstr(Instr instr) {
         if (needsLineNumInfo) {
             needsLineNumInfo = false;
-            addInstr(manager.newLineNumber(_lastProcessedLineNum));
+
+            if (needsCodeCoverage()) {
+                addInstr(new LineNumberInstr(_lastProcessedLineNum, true, (coverageMode & CoverageData.ONESHOT_LINES) != 0));
```

Comment:
This is an implicit true in constructor isn't it?

Reply:
So you are saying remove that true since we'll always provide oneshot = true | false, which disambiguates enough?

Reply:
I modified the code to pass the coverage mode instead of booleans.